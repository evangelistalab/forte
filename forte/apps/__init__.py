from .mcscf import run_mcscf
from .hf import run_hf
