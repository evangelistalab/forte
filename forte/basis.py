class Basis:
    """
    A class used to store basis information.
    """
    def __init__(self, basis):
        """
        initialize a Basis object

        Parameters
        ----------
        basis : str
            a basis object
        """
        self._basis = basis

    def __repr__(self):
        """
        return a string representation of this object
        """
        return f'Basis(\'{self._basis}\')'

    def __str__(self):
        """
        return a string representation of this object
        """
        return self._basis
