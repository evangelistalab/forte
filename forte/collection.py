class Collection:
    """
    A class to that handles forte objects

    Attributes
    ----------
    """
    def __init__(self, scf_info=None, mospaceinfo=None, ints=None, psi_wfn=None):
        """
        initialize a Basis object

        Parameters
        ----------
        restricted : bool
            a basis object
        """
        self._collection = {}
        self._collection['scf_info'] = scf_info
        self._collection['mospaceinfo'] = mospaceinfo
        self._collection['ints'] = ints
        self._collection['psi_wfn'] = psi_wfn

    def __repr__(self):
        """
        return a string representation of this object
        """
        return f'Collection()'

    def __str__(self):
        """
        return a string representation of this object
        """
        return f'Collection()'

    @property
    def scf_info(self):
        return self._collection['scf_info']

    @scf_info.setter
    def scf_info(self, val):
        self._collection['scf_info'] = val

    @property
    def mospaceinfo(self):
        return self._collection['mospaceinfo']

    @property
    def ints(self):
        return self._collection['ints']

    @property
    def psi_wfn(self):
        return self._collection['psi_wfn']

    @psi_wfn.setter
    def psi_wfn(self, val):
        self._collection['psi_wfn'] = val
