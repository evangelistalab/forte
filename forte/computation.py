# from .results import Results

# class Computation:
#     """
#     A class used to represent a computation.

#     Attributes
#     ----------
#     basis : str
#         the basis name
#     """
#     def __init__(self, model=None, method=None):
#         """
#         initialize a Basis object

#         Parameters
#         ----------
#         basis : str
#             a basis object
#         """
#         self._model = model
#         self._method = method

#     def __repr__(self):
#         """
#         return a string representation of this object
#         """
#         return f'Computation(model={repr(self._model)},method={repr(self._method)})'

#     def __str__(self):
#         """
#         return a string representation of this object
#         """
#         return self._basis

#     @property
#     def model(self):
#         return self._model

#     @property
#     def method(self):
#         return self._method

#     def energy(self):
#         """Compute the energy."""
#         results = Results()
#         self._method.energy(self._model, results=results)
#         return results
