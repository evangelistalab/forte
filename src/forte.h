/*
 * @BEGIN LICENSE
 *
 * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
 * that implements a variety of quantum chemistry methods for strongly
 * correlated electrons.
 *
 * Copyright (c) 2012-2019 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
 *
 * The copyrights for code used from other parties are included in
 * the corresponding files.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see http://www.gnu.org/licenses/.
 *
 * @END LICENSE
 */

#include "base_classes/forte_options.h"

namespace forte {

class ForteIntegrals;

void forte_options(ForteOptions& options);

std::pair<int, int> startup();
void banner();
void cleanup();

void read_options(ForteOptions& options);
psi::SharedWavefunction run_forte(psi::SharedWavefunction ref_wfn, psi::Options& options);

std::shared_ptr<MOSpaceInfo> make_mo_space_info(psi::SharedWavefunction ref_wfn,
                                                std::shared_ptr<ForteOptions> options);

psi::SharedMatrix make_aosubspace_projector(psi::SharedWavefunction ref_wfn, psi::Options& options);

void make_ci_nos(psi::SharedWavefunction ref_wfn, psi::Options& options,
                 std::shared_ptr<ForteIntegrals> ints, std::shared_ptr<MOSpaceInfo> mo_space_info);

double forte_old_methods(psi::SharedWavefunction ref_wfn, psi::Options& options,
                         std::shared_ptr<ForteIntegrals> ints,
                         std::shared_ptr<MOSpaceInfo> mo_space_info);
} // namespace forte
