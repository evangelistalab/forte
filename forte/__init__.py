#
# @BEGIN LICENSE
#
# Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
# that implements a variety of quantum chemistry methods for strongly
# correlated electrons.
#
# Copyright (c) 2012-2017 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
#
# The copyrights for code used from other parties are included in
# the corresponding files.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program.  If not, see http://www.gnu.org/licenses/.
#
# @END LICENSE
#
"""Plugin docstring.

"""

import sys
import logging

logging.basicConfig(filename='forte.log', level=logging.INFO)
logging.info('Starting the Forte logger')

# Load Python modules
from .molecule import Molecule
from .basis import Basis
from .results import Results
from .model import Model, MolecularModel

from .pymodule import *
from .register_forte_options import *
from .core import ForteManager, clean_options
from .forte import *

__version__ = '0.2.0'
__author__ = 'Forte Developers'

# Create a ForteOptions object (stores all options)
forte_options = forte.ForteOptions()

# Register options defined in Forte in the forte_options object
register_forte_options(forte_options)

# create a singleton to handle startup and cleanup of forte
ForteManager()

# If we are running psi4, push the options defined in forte_options to psi
if 'psi4' in sys.modules:
    psi_options = psi4.core.get_options()
    psi_options.set_current_module('FORTE')
    forte_options.push_options_to_psi4(psi_options)
