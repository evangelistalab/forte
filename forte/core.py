#
# @BEGIN LICENSE
#
# Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
# that implements a variety of quantum chemistry methods for strongly
# correlated electrons.
#
# Copyright (c) 2012-2021 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
#
# The copyrights for code used from other parties are included in
# the corresponding files.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
# @END LICENSE
#

import logging
import psi4
import forte

logging_depth = 0


def clean_options():
    """
    A function to clear the options object

    This function does the following
    1. clears the psi4 and Forte options object
    2. re-registers all of Forte options (in their default values)
    3. pushes the value of Forte options to the psi4 options object
    """
    # clear options and allocate a fresh forte.forte_options object
    psi4.core.clean_options()
    forte.forte_options = forte.ForteOptions()

    # register options defined in Forte in the forte_options object
    forte.register_forte_options(forte.forte_options)

    # Push the options defined in forte_options to psi
    psi_options = psi4.core.get_options()
    psi_options.set_current_module('FORTE')
    forte.forte_options.push_options_to_psi4(psi_options)


class ForteManager(object):
    """Singleton class to handle startup and cleanup of forte (mostly ambit)"""
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = True
            # Put any initialization here.
            my_proc, n_nodes = forte.startup()
        return cls._instance

    def __del__(cls):
        forte.cleanup()


def start_logging(filename='forte.log', level=logging.DEBUG):
    """
    This function sets the output of logs to ``filename`` (default = forte.log)
    and sets the log level to all information.

    Parameters
    ----------
    filename: str
        the name of the log file (default = 'forte.log')
    level: {logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR, logging.CRITICAL}
        the level of severity of the events tracked (default = logging.DEBUG, which means track everything)
    """
    logging.basicConfig(filename=filename, level=level, format='# %(asctime)s | %(levelname)s | %(message)s')
    logging.info('Starting the Forte logger')


def flog(level, msg):
    """
    Log the message ``msg`` with logging level ``level``.

    ``level`` should be chosen in this way:
    debug: for detailed ouput mostly for diagnostic purpose
    info: for information produced during normal operation of the program
    warning: for a warning regarding a particular runtime event
    error: for an error that does not raise an exception

    Parameters
    ----------
    level: str
        the level of the message logged. Can be any of ('debug','info','warning','error')
    msg: str
        the text to be logged

    This function calls the logging module and puts some spaces before
    the message that we want to log.
    """
    global logging_depth
    level = level.lower()
    spaces = ' ' * max((logging_depth - 1), 0) * 2
    s = f"{spaces}{msg}"
    if level == 'info':
        logging.info(s)
    elif level == 'warning':
        logging.warning(s)
    elif level == 'debug':
        logging.debug(s)
    elif level == 'error':
        logging.error(s)
    else:
        raise ValueError(f'forte.core.flog was called with an unrecognized level ({level})')


def increase_log_depth(func):
    """
    This is a decorator used to increase the depth of forte's log.

    It is used to decorate the run() function of solvers:
        @increase_log_dept
            def run(self):
                ...
    """
    def wrapper(*args, **kwargs):
        global logging_depth
        logging_depth += 1
        func(*args, **kwargs)
        logging_depth += -1

    return wrapper
