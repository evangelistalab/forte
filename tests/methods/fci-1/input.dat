#! Generated using commit GITCOMMIT

import forte

refscf = -14.54873910108353
reffci = -14.595808852754054

molecule {
0 1
Li
Li 1 R
R = 3.0
units bohr
}

set {
  basis sto-3g
  scf_type pk
  e_convergence 12
}

set forte {
  job_type fci
}

energy('scf')
compare_values(refscf, variable("CURRENT ENERGY"),11, "SCF energy") #TEST

energy('forte')
compare_values(reffci, variable("CURRENT ENERGY"),11, "FCI energy") #TEST
