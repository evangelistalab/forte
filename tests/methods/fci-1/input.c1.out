
    -----------------------------------------------------------------------
          Psi4: An Open-Source Ab Initio Electronic Structure Package
                               Psi4 undefined 

                         Git: Rev {master} 2feab68 


    D. G. A. Smith, L. A. Burns, A. C. Simmonett, R. M. Parrish,
    M. C. Schieber, R. Galvelis, P. Kraus, H. Kruse, R. Di Remigio,
    A. Alenaizan, A. M. James, S. Lehtola, J. P. Misiewicz, M. Scheurer,
    R. A. Shaw, J. B. Schriber, Y. Xie, Z. L. Glick, D. A. Sirianni,
    J. S. O'Brien, J. M. Waldrop, A. Kumar, E. G. Hohenstein,
    B. P. Pritchard, B. R. Brooks, H. F. Schaefer III, A. Yu. Sokolov,
    K. Patkowski, A. E. DePrince III, U. Bozkaya, R. A. King,
    F. A. Evangelista, J. M. Turney, T. D. Crawford, C. D. Sherrill,
    J. Chem. Phys. 152(18) 184108 (2020). https://doi.org/10.1063/5.0006002

                            Additional Code Authors
    E. T. Seidl, C. L. Janssen, E. F. Valeev, M. L. Leininger,
    J. F. Gonthier, R. M. Richard, H. R. McAlexander, M. Saitow, X. Wang,
    P. Verma, M. H. Lechner, and A. Jiang

             Previous Authors, Complete List of Code Contributors,
                       and Citations for Specific Modules
    https://github.com/psi4/psi4/blob/master/codemeta.json
    https://github.com/psi4/psi4/graphs/contributors
    http://psicode.org/psi4manual/master/introduction.html#citing-psifour

    -----------------------------------------------------------------------


    Psi4 started on: Tuesday, 21 December 2021 04:07PM

    Process ID: 93395
    Host:       CHE-FEVANGE
    PSIDATADIR: /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4
    Memory:     500.0 MiB
    Threads:    1
    
  ==> Input File <==

--------------------------------------------------------------------------
#! Generated using commit GITCOMMIT

import forte

refscf = -14.54873910108353
reffci = -14.595808852754054

molecule {
0 1
Li
Li 1 R
R = 3.0
units bohr
symmetry c1
}

set {
  basis sto-3g
  scf_type pk
  e_convergence 12
}

set forte {
  active_space_solver fci 
}

energy('scf')
compare_values(refscf, variable("CURRENT ENERGY"),11, "SCF energy") #TEST

energy('forte')
compare_values(reffci, variable("CURRENT ENERGY"),11, "FCI energy") #TEST
--------------------------------------------------------------------------

Scratch directory: /tmp/

*** tstart() called on CHE-FEVANGE
*** at Tue Dec 21 16:07:11 2021

   => Loading Basis Set <=

    Name: STO-3G
    Role: ORBITAL
    Keyword: BASIS
    atoms 1-2 entry LI         line    31 file /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4/basis/sto-3g.gbs 


         ---------------------------------------------------------
                                   SCF
               by Justin Turney, Rob Parrish, Andy Simmonett
                          and Daniel G. A. Smith
                              RHF Reference
                        1 Threads,    500 MiB Core
         ---------------------------------------------------------

  ==> Geometry <==

    Molecular point group: c1
    Full point group: D_inf_h

    Geometry (in Bohr), charge = 0, multiplicity = 1:

       Center              X                  Y                   Z               Mass       
    ------------   -----------------  -----------------  -----------------  -----------------
         LI           0.000000000000     0.000000000000    -1.500000000000     7.016003436600
         LI           0.000000000000     0.000000000000     1.500000000000     7.016003436600

  Running in c1 symmetry.

  Rotational constants: A = ************  B =      1.90674  C =      1.90674 [cm^-1]
  Rotational constants: A = ************  B =  57162.68592  C =  57162.68592 [MHz]
  Nuclear repulsion =    3.000000000000000

  Charge       = 0
  Multiplicity = 1
  Electrons    = 6
  Nalpha       = 3
  Nbeta        = 3

  ==> Algorithm <==

  SCF Algorithm Type is PK.
  DIIS enabled.
  MOM disabled.
  Fractional occupation disabled.
  Guess Type is SAD.
  Energy threshold   = 1.00e-12
  Density threshold  = 1.00e-06
  Integral threshold = 1.00e-12

  ==> Primary Basis <==

  Basis Set: STO-3G
    Blend: STO-3G
    Number of shells: 6
    Number of basis functions: 10
    Number of Cartesian functions: 10
    Spherical Harmonics?: true
    Max angular momentum: 1

  ==> Integral Setup <==

  Using in-core PK algorithm.
   Calculation information:
      Number of atoms:                   2
      Number of AO shells:               6
      Number of primitives:             18
      Number of atomic orbitals:        10
      Number of basis functions:        10

      Integral cutoff                 1.00e-12
      Number of threads:                 1

  Performing in-core PK
  Using 3080 doubles for integral storage.
  We computed 231 shell quartets total.
  Whereas there are 231 unique shell quartets.

  ==> DiskJK: Disk-Based J/K Matrices <==

    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Memory [MiB]:              375
    Schwarz Cutoff:          1E-12

    OpenMP threads:              1

  Minimum eigenvalue in the overlap matrix is 1.1364559165E-02.
  Reciprocal condition number of the overlap matrix is 5.2366670220E-03.
    Using symmetric orthogonalization.

  ==> Pre-Iterations <==

  SCF Guess: Superposition of Atomic Densities via on-the-fly atomic UHF (no occupation information).

   -------------------------
    Irrep   Nso     Nmo    
   -------------------------
     A         10      10 
   -------------------------
    Total      10      10
   -------------------------

  ==> Iterations <==

                        Total Energy        Delta E     RMS |[F,P]|

   @RHF iter SAD:   -14.35526221907790   -1.43553e+01   0.00000e+00 
   @RHF iter   1:   -14.54744094710718   -1.92179e-01   2.46202e-03 DIIS
   @RHF iter   2:   -14.54852099240109   -1.08005e-03   7.49418e-04 DIIS
   @RHF iter   3:   -14.54871577585422   -1.94783e-04   2.57157e-04 DIIS
   @RHF iter   4:   -14.54873909938384   -2.33235e-05   2.82987e-06 DIIS
   @RHF iter   5:   -14.54873910108192   -1.69807e-09   7.75893e-08 DIIS
   @RHF iter   6:   -14.54873910108352   -1.59872e-12   3.08157e-09 DIIS
   @RHF iter   7:   -14.54873910108352   -1.77636e-15   8.29755e-12 DIIS
  Energy and wave function converged.


  ==> Post-Iterations <==

    Orbital Energies [Eh]
    ---------------------

    Doubly Occupied:                                                      

       1A     -2.406336     2A     -2.377194     3A     -0.171891  

    Virtual:                                                              

       4A      0.078832     5A      0.078832     6A      0.088519  
       7A      0.200395     8A      0.296222     9A      0.296222  
      10A      0.859229  

    Final Occupation by Irrep:
              A 
    DOCC [     3 ]

  @RHF Final Energy:   -14.54873910108352

   => Energetics <=

    Nuclear Repulsion Energy =              3.0000000000000000
    One-Electron Energy =                 -25.1866354005934951
    Two-Electron Energy =                   7.6378962995099773
    Total Energy =                        -14.5487391010835179

Computation Completed


Properties will be evaluated at   0.000000,   0.000000,   0.000000 [a0]

Properties computed using the SCF density matrix

  Nuclear Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:     0.0000

  Electronic Dipole Moment: [e a0]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000

  Dipole Moment: [e a0]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000     Total:     0.0000

  Dipole Moment: [D]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000     Total:     0.0000


*** tstop() called on CHE-FEVANGE at Tue Dec 21 16:07:12 2021
Module time:
	user time   =       0.31 seconds =       0.01 minutes
	system time =       0.01 seconds =       0.00 minutes
	total time  =          1 seconds =       0.02 minutes
Total time:
	user time   =       0.31 seconds =       0.01 minutes
	system time =       0.01 seconds =       0.00 minutes
	total time  =          1 seconds =       0.02 minutes
    SCF energy............................................................................PASSED

Scratch directory: /tmp/

  Forte
  ----------------------------------------------------------------------------
  A suite of quantum chemistry methods for strongly correlated electrons

    git branch: make_so_ints - git commit: 8aa9a2e8

  Developed by:
    Francesco A. Evangelista, Chenyang Li, Kevin P. Hannon,
    Jeffrey B. Schriber, Tianyuan Zhang, Chenxi Cai,
    Nan He, Nicholas Stair, Shuhe Wang, Renke Huang
  ----------------------------------------------------------------------------

  Size of Determinant class: 128 bits

  Preparing forte objects from a Psi4 Wavefunction object
  No reference wave function provided for Forte. Computing SCF orbitals using Psi4 ...

*** tstart() called on CHE-FEVANGE
*** at Tue Dec 21 16:07:12 2021

   => Loading Basis Set <=

    Name: STO-3G
    Role: ORBITAL
    Keyword: BASIS
    atoms 1-2 entry LI         line    31 file /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4/basis/sto-3g.gbs 


         ---------------------------------------------------------
                                   SCF
               by Justin Turney, Rob Parrish, Andy Simmonett
                          and Daniel G. A. Smith
                              RHF Reference
                        1 Threads,    500 MiB Core
         ---------------------------------------------------------

  ==> Geometry <==

    Molecular point group: c1
    Full point group: D_inf_h

    Geometry (in Bohr), charge = 0, multiplicity = 1:

       Center              X                  Y                   Z               Mass       
    ------------   -----------------  -----------------  -----------------  -----------------
         LI           0.000000000000     0.000000000000    -1.500000000000     7.016003436600
         LI           0.000000000000     0.000000000000     1.500000000000     7.016003436600

  Running in c1 symmetry.

  Rotational constants: A = ************  B =      1.90674  C =      1.90674 [cm^-1]
  Rotational constants: A = ************  B =  57162.68592  C =  57162.68592 [MHz]
  Nuclear repulsion =    3.000000000000000

  Charge       = 0
  Multiplicity = 1
  Electrons    = 6
  Nalpha       = 3
  Nbeta        = 3

  ==> Algorithm <==

  SCF Algorithm Type is PK.
  DIIS enabled.
  MOM disabled.
  Fractional occupation disabled.
  Guess Type is SAD.
  Energy threshold   = 1.00e-12
  Density threshold  = 1.00e-08
  Integral threshold = 1.00e-12

  ==> Primary Basis <==

  Basis Set: STO-3G
    Blend: STO-3G
    Number of shells: 6
    Number of basis functions: 10
    Number of Cartesian functions: 10
    Spherical Harmonics?: true
    Max angular momentum: 1

  ==> Integral Setup <==

  Using in-core PK algorithm.
   Calculation information:
      Number of atoms:                   2
      Number of AO shells:               6
      Number of primitives:             18
      Number of atomic orbitals:        10
      Number of basis functions:        10

      Integral cutoff                 1.00e-12
      Number of threads:                 1

  Performing in-core PK
  Using 3080 doubles for integral storage.
  We computed 231 shell quartets total.
  Whereas there are 231 unique shell quartets.

  ==> DiskJK: Disk-Based J/K Matrices <==

    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Memory [MiB]:              375
    Schwarz Cutoff:          1E-12

    OpenMP threads:              1

  Minimum eigenvalue in the overlap matrix is 1.1364559165E-02.
  Reciprocal condition number of the overlap matrix is 5.2366670220E-03.
    Using symmetric orthogonalization.

  ==> Pre-Iterations <==

  SCF Guess: Superposition of Atomic Densities via on-the-fly atomic UHF (no occupation information).

   -------------------------
    Irrep   Nso     Nmo    
   -------------------------
     A         10      10 
   -------------------------
    Total      10      10
   -------------------------

  ==> Iterations <==

                        Total Energy        Delta E     RMS |[F,P]|

   @RHF iter SAD:   -14.35526221907790   -1.43553e+01   0.00000e+00 
   @RHF iter   1:   -14.54744094710718   -1.92179e-01   2.46202e-03 DIIS
   @RHF iter   2:   -14.54852099240109   -1.08005e-03   7.49418e-04 DIIS
   @RHF iter   3:   -14.54871577585422   -1.94783e-04   2.57157e-04 DIIS
   @RHF iter   4:   -14.54873909938384   -2.33235e-05   2.82987e-06 DIIS
   @RHF iter   5:   -14.54873910108192   -1.69807e-09   7.75893e-08 DIIS
   @RHF iter   6:   -14.54873910108352   -1.59872e-12   3.08157e-09 DIIS
   @RHF iter   7:   -14.54873910108352   -3.55271e-15   8.29755e-12 DIIS
  Energy and wave function converged.


  ==> Post-Iterations <==

    Orbital Energies [Eh]
    ---------------------

    Doubly Occupied:                                                      

       1A     -2.406336     2A     -2.377194     3A     -0.171891  

    Virtual:                                                              

       4A      0.078832     5A      0.078832     6A      0.088519  
       7A      0.200395     8A      0.296222     9A      0.296222  
      10A      0.859229  

    Final Occupation by Irrep:
              A 
    DOCC [     3 ]

  @RHF Final Energy:   -14.54873910108352

   => Energetics <=

    Nuclear Repulsion Energy =              3.0000000000000000
    One-Electron Energy =                 -25.1866354005934951
    Two-Electron Energy =                   7.6378962995099755
    Total Energy =                        -14.5487391010835196

Computation Completed


Properties will be evaluated at   0.000000,   0.000000,   0.000000 [a0]

Properties computed using the SCF density matrix

  Nuclear Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:     0.0000

  Electronic Dipole Moment: [e a0]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000

  Dipole Moment: [e a0]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000     Total:     0.0000

  Dipole Moment: [D]
     X:    -0.0000      Y:    -0.0000      Z:    -0.0000     Total:     0.0000


*** tstop() called on CHE-FEVANGE at Tue Dec 21 16:07:12 2021
Module time:
	user time   =       0.28 seconds =       0.00 minutes
	system time =       0.01 seconds =       0.00 minutes
	total time  =          0 seconds =       0.00 minutes
Total time:
	user time   =       0.65 seconds =       0.01 minutes
	system time =       0.02 seconds =       0.00 minutes
	total time  =          1 seconds =       0.02 minutes


  ==> MO Space Information <==

  -------------------------------
                        A   Sum
  -------------------------------
    FROZEN_DOCC         0     0
    RESTRICTED_DOCC     0     0
    GAS1               10    10
    GAS2                0     0
    GAS3                0     0
    GAS4                0     0
    GAS5                0     0
    GAS6                0     0
    RESTRICTED_UOCC     0     0
    FROZEN_UOCC         0     0
    Total              10    10
  -------------------------------   => Loading Basis Set <=

    Name: STO-3G
    Role: ORBITAL
    Keyword: MINAO_BASIS
    atoms 1-2 entry LI         line    31 file /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4/basis/sto-3g.gbs 


  ==> List of Planes Requested <==

  Forte will use psi4 integrals

  ==> Primary Basis Set Summary <==

  Basis Set: STO-3G
    Blend: STO-3G
    Number of shells: 6
    Number of basis functions: 10
    Number of Cartesian functions: 10
    Spherical Harmonics?: true
    Max angular momentum: 1


  JK created using conventional PK integrals
  Using in-core PK algorithm.
   Calculation information:
      Number of atoms:                   2
      Number of AO shells:               6
      Number of primitives:             18
      Number of atomic orbitals:        10
      Number of basis functions:        10

      Integral cutoff                 1.00e-12
      Number of threads:                 1

  Performing in-core PK
  Using 3080 doubles for integral storage.
  We computed 231 shell quartets total.
  Whereas there are 231 unique shell quartets.

  ==> DiskJK: Disk-Based J/K Matrices <==

    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Memory [MiB]:              400
    Schwarz Cutoff:          1E-12

    OpenMP threads:              1



  ==> Integral Transformation <==

  Number of molecular orbitals:                         10
  Number of correlated molecular orbitals:              10
  Number of frozen occupied orbitals:                    0
  Number of frozen unoccupied orbitals:                  0
  Two-electron integral type:                 Conventional


  Computing Conventional Integrals	Presorting SO-basis two-electron integrals.
	Sorting File: SO Ints (nn|nn) nbuckets = 1
	Constructing frozen core operators
	Starting first half-transformation.
	Sorting half-transformed integrals.
	First half integral transformation complete.
	Starting second half-transformation.
	Two-electron integral transformation complete.

  Integral transformation done. 0.01299899 s
  Reading the two-electron integrals from disk
  Size of two-electron integrals:   0.000224 GB
  Timing for conventional integral transformation:            0.041 s.
  Timing for freezing core and virtual orbitals:              0.000 s.
  Timing for computing conventional integrals:                0.041 s.

  ==> Summary of Active Space Solver Input <==

    Irrep.  Multi.(2ms)      N
    --------------------------
        A      1  (  0)      1
    --------------------------
    N: number of roots
    ms: spin z component
    Total number of roots:   1
    --------------------------


  ==> String Lists <==

  Number of alpha electrons     = 3
  Number of beta electrons      = 3
  Number of alpha strings       = 120
  Number of beta strings        = 120
  Number of alpha strings (N-3) = 1
  Number of beta strings (N-3)  = 1
  Timing for strings        =      0.000 s
  Timing for NN strings     =      0.000 s
  Timing for VO strings     =      0.002 s
  Timing for OO strings     =      0.000 s
  Timing for VVOO strings   =      0.015 s
  Timing for VOVO strings   =      0.000 s
  Timing for 1-hole strings =      0.000 s
  Timing for 2-hole strings =      0.001 s
  Timing for 3-hole strings =      0.003 s
  Total timing              =      0.018 s

  ==> FCI Solver <==

    Number of determinants                       14400
    Symmetry                                         0
    Multiplicity                                     1
    Number of roots                                  1
    Target root                                      0
    Trial vectors per root                          10

  Allocating memory for the Hamiltonian algorithm. Size: 2 x 120 x 120.   Memory: 0.000215 GB
  Initial guess space is incomplete.
  Adding 1 determinant(s).

  ==> FCI Initial Guess <==

  ---------------------------------------------
    Root            Energy     <S^2>   Spin
  ---------------------------------------------
      0      -14.588934410108  0.000  singlet
      1      -14.562586453491  2.000  triplet
      2      -14.562586453491  2.000  triplet
      3      -14.524068499000  2.000  triplet
      4      -14.488403359830  2.000  triplet
      5      -14.462133305023  0.000  singlet
      6      -14.462133305023  0.000  singlet
      7      -14.455080858784  0.000  singlet
      8      -14.455080858784  0.000  singlet
      9      -14.450449464563  2.000  triplet
     10      -14.450449464563  2.000  triplet
     11      -14.449356583139  2.000  triplet
     12      -14.418208330514  0.000  singlet
     13      -14.414687042212  0.000  singlet
     14      -14.414687042212  0.000  singlet
     15      -14.413988427680  0.000  singlet
     16      -14.389241650050  2.000  triplet
     17      -14.389241650050  2.000  triplet
     18      -14.357374935156  0.000  singlet
     19      -14.357374935156  0.000  singlet
     20      -14.355202244479  0.000  singlet
  ---------------------------------------------
  Timing for initial guess  =      0.002 s

  Projecting out root 1
  Projecting out root 2
  Projecting out root 3
  Projecting out root 4
  Projecting out root 9
  Projecting out root 10
  Projecting out root 11
  Projecting out root 16
  Projecting out root 17

  ==> Diagonalizing Hamiltonian <==

  Energy   convergence: 1.00e-12
  Residual convergence: 1.00e-06
  -----------------------------------------------------
    Iter.      Avg. Energy       Delta_E     Res. Norm
  -----------------------------------------------------
      1      -14.588934410108  -1.459e+01  +7.691e-02
      2      -14.594848517228  -5.914e-03  +2.086e-02
      3      -14.595675528946  -8.270e-04  +1.099e-02
      4      -14.595794557237  -1.190e-04  +3.883e-03
      5      -14.595808464997  -1.391e-05  +9.319e-04
      6      -14.595808849144  -3.841e-07  +1.138e-04
      7      -14.595808852699  -3.555e-09  +1.423e-05
      8      -14.595808852753  -5.415e-11  +1.730e-06
      9      -14.595808852754  -8.029e-13  +2.365e-07
  -----------------------------------------------------
  The Davidson-Liu algorithm converged in 10 iterations.

  ==> Root No. 0 <==

    2220000000     -0.89740851
    2200200000      0.29206212
    2202000000      0.29206212
    2200020000      0.14391930

    Total Energy:     -14.595808852754, <S^2>: 0.000000

  ==> Energy Summary <==

    Multi.(2ms)  Irrep.  No.               Energy      <S^2>
    --------------------------------------------------------
       1  (  0)     A     0      -14.595808852754   0.000000
    --------------------------------------------------------

  Time to prepare integrals:        0.067 seconds
  Time to run job          :        0.252 seconds
  Total                    :        0.319 seconds
    FCI energy............................................................................PASSED

    Psi4 stopped on: Tuesday, 21 December 2021 04:07PM
    Psi4 wall time for execution: 0:00:01.59

*** Psi4 exiting successfully. Buy a developer a beer!
